horovod/spark/lightning/remote.py

```diff
@@ -128,11 +128,16 @@ def train(serialized_model):
             # Lightning requires to add checkpoint callbacks for all ranks.
             # Otherwise we are seeing hanging in training.
             _checkpoint_callback = checkpoint_callback
```

Comment:
For simplicity, just relying on this `_checkpoint_callback` variable != None check is enough.
If `_checkpoint_callback` is not None: modify `dir_path` and `filename`, and `callbacks.append(_checkpoint_callback)`

Reply:
if _checkpoint_callback is None, it will set to default ModelCheckpoint in line 141.
If user do not pass in a checkpoint callback, we still want to add the default `ModelCheckpoint` in to callback list, right?